hack/update-api-resource-list.sh

```diff
@@ -0,0 +1,114 @@
+AUTH_TOKEN_HEADER="Authorization: Bearer dummy_token"
+
+function fetch_api_resource()
+{
+for RESOURCE in $(curl -s --cacert $KUBE_APISERVER_CERT -H "$AUTH_TOKEN_HEADER" $APISERVER_URL/api/v1 | jq -c .resources[]); do
+  NAME=$(echo $RESOURCE | jq .name | tr -d '"')
+  NAMESPACED=$(echo $RESOURCE | jq .namespaced)
+  for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+    echo ,v1,$NAME,$NAMESPACED,$VERB
+  done
+done
+
+# List group versions
+for GROUP_VERSION in $(curl -s --cacert $KUBE_APISERVER_CERT -H "$AUTH_TOKEN_HEADER" $APISERVER_URL/apis | jq .groups[].versions[].groupVersion | tr -d '"'); do
+  # List resources
+  for RESOURCE in $(curl -s --cacert $KUBE_APISERVER_CERT -H "$AUTH_TOKEN_HEADER" $APISERVER_URL/apis/$GROUP_VERSION | jq -c '.resources[]'); do
+    NAME=$(echo $RESOURCE | jq '.name' | tr -d '"')
+    NAMESPACED=$(echo $RESOURCE | jq '.namespaced')
+    for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+      echo $(echo $GROUP_VERSION | tr '/' ','),$NAME,$NAMESPACED,$VERB
+    done
+  done
+done
+}
+
+kube::util::trap_add cleanup EXIT SIGINT
+
+kube::golang::setup_env
+
+apiserver=$(kube::util::find-binary "kube-apiserver")
+
+ETCD_HOST=${ETCD_HOST:-127.0.0.1}
+ETCD_PORT=${ETCD_PORT:-2379}
+API_PORT=${API_PORT:-8050}
+API_HOST=${API_HOST:-127.0.0.1}
+
+kube::etcd::start
+
+echo "dummy_token,admin,admin" > $TMP_DIR/tokenauth.csv
+
+# Start kube-apiserver
+kube::log::status "Starting kube-apiserver"
+"${KUBE_OUTPUT_HOSTBIN}/kube-apiserver" \
+  --bind-address="${API_HOST}" \
+  --secure-port="${API_PORT}" \
+  --etcd-servers="http://${ETCD_HOST}:${ETCD_PORT}" \
+  --advertise-address="10.10.10.10" \
+  --cert-dir="${TMP_DIR}/certs" \
+  --runtime-config="api/all=true" \
+  --token-auth-file=$TMP_DIR/tokenauth.csv \
+  --logtostderr \
+  --v=2 \
+  --service-cluster-ip-range="10.0.0.0/24" >/tmp/openapi-api-server.log 2>&1 &
+APISERVER_PID=$!
+
+kube::util::wait_for_url_ssl "https://${API_HOST}:${API_PORT}/healthz" "apiserver: " 1 30 1 --cacert $KUBE_APISERVER_CERT -H "$AUTH_TOKEN_HEADER"
+
+APISERVER_URL="https://${API_HOST}:${API_PORT}"
+
+KUBE_RESOURCE_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources_all.csv"
+KUBE_RESOURCE_WHITELIST_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources_whitelist.csv"
+KUBE_RESOURCE_TEST_FILE="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource/resources.csv"
+
+# Make sure subresource come after parent resource
+fetch_api_resource | sort -t',' -k1,1 -k2,2 -k3,3 -k4 -o ${KUBE_RESOURCE_FILE}
+sort -t',' -k1,1 -k2,2 -k3,3 -k4 ${KUBE_RESOURCE_WHITELIST_FILE} -o ${KUBE_RESOURCE_WHITELIST_FILE}
+# GNU comm and BSD comm have different defaulting. We sort the inputs to satisfy 
+# the defaulting first, then sort the output to more readable csv format
+comm -13 <(sort ${KUBE_RESOURCE_WHITELIST_FILE}) <(sort ${KUBE_RESOURCE_FILE}) | sort -t',' -k1,1 -k2,2 -k3,3 -k4 -o ${KUBE_RESOURCE_TEST_FILE}
```

Comment:
Add a comment here explaining the intent, which is to remove whitelist lines from RESOURCE_FILE